frame/bounties/src/lib.rs

```diff
@@ -653,12 +780,596 @@ decl_module! {
+			#[compact] subbounty_id: BountyIndex,
+			subcurator: <T::Lookup as StaticLookup>::Source,
+			#[compact] fee: BalanceOf<T>,
+		) {
+			let signer = ensure_signed(origin)?;
+			let subcurator = T::Lookup::lookup(subcurator)?;
+
+			// Ensure parent bounty is Active & get status of curator
+			let ( master_curator, _) = Self::ensure_bounty_active(bounty_id)?;
+
+			// Mutate the Subbounty instance
+			SubBounties::<T>::try_mutate_exists(bounty_id,
+				subbounty_id,
+				|maybe_subbounty| -> DispatchResult {
+
+				let mut subbounty = maybe_subbounty
+					.as_mut()
+					.ok_or(Error::<T>::InvalidIndex)?;
+
+				// Ensure sure caller is curator
+				ensure!( signer == master_curator, Error::<T>::RequireCurator);
+
+				// Ensure subbounty is in expected state
+				match subbounty.status {
+					SubBountyStatus::Added => {},
+					_ => return Err(Error::<T>::UnexpectedStatus.into()),
+				};
+
+				// if curator self assign as subcurator, fee is 0
+				if master_curator != subcurator {
+					ensure!(fee < subbounty.value, Error::<T>::InvalidFee);
+					subbounty.fee = fee;
+				} else {
+					subbounty.fee = Zero::zero();
+				}
+				// update the subbounty state
+				subbounty.status = SubBountyStatus::SubCuratorProposed {
+					subcurator: subcurator
+				};
+				Ok(())
+			})?;
+		}
+
+		/// Accept the subcurator role for the subbounty.
+		///
+		/// A deposit will be reserved from subcurator and
+		/// refund upon successful payout or cancellation.
+		/// In case if "master curator" is "subcurator",
+		/// curator deposit is Zero or ignored.
+		///
+		/// The dispatch origin for this call must be
+		/// the subcurator of this subbounty.
+		///
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work.
+		///
+		/// Subbounty must be in "CuratorProposed" state, for
+		/// processing the call. and state of subbounty is
+		/// moved to "Active" on successful call
+		/// completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		#[weight = 10_000]
+		fn accept_subcurator(origin,
+			#[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+		) {
+			let signer = ensure_signed(origin)?;
+
+			// Ensure parent bounty is Active & get status of curator
+			let ( master_curator, _) = Self::ensure_bounty_active(bounty_id)?;
+
+			// Mutate Subbounty
+			SubBounties::<T>::try_mutate_exists(bounty_id, subbounty_id,
+				|maybe_subbounty| -> DispatchResult {
+
+				let mut subbounty = maybe_subbounty
+					.as_mut()
+					.ok_or(Error::<T>::InvalidIndex)?;
+
+				// Ensure subbounty is in expected state
+				match subbounty.status {
+					SubBountyStatus::SubCuratorProposed { ref subcurator } => {
+						ensure!(signer == *subcurator, Error::<T>::RequireCurator);
+						// TODO :: Have to recheck
+						// if `master curator` is `subcurator`
+						// curator deposit is ignored or Zero.
+						if master_curator != *subcurator {
```

Comment:
I think we don't need this. Just to keep it simple.

Reply:
Do we need to consider, "curator deposit" even if master curator self assign as subcurator ?